src/java/htsjdk/samtools/SAMFileWriterFactory.java

```diff
@@ -311,28 +332,91 @@ public CRAMFileWriter makeCRAMWriter(final SAMFileHeader header, final OutputStr
         return writer;
     }
 
+    /**
+     * Create a CRAMFileWriter on an output file. Requires input record to be presorted to match the
+     * sort order defined by the input header.
+     *
+     * Note: does not honor factory settings for USE_ASYNC_IO.
+     * Read names and tags are preserved.
+     *
+     * @param header entire header. Sort order is determined by the sortOrder property of this arg.
+     * @param outputFile where to write the output.  Must end with .sam, .bam or .cram.
+     * @param referenceFasta reference sequence file
+     * @return CRAMFileWriter
+     *
+     */
     public CRAMFileWriter makeCRAMWriter(final SAMFileHeader header, final File outputFile, final File referenceFasta) {
+        return createCRAMWriter(header, true, outputFile, referenceFasta);
+    }
 
-        final boolean createIndex = this.createIndex && IOUtil.isRegularPath(outputFile);
-        if (this.createIndex && !createIndex) {
-            System.err.println("Cannot create index for CAM because output file is not a regular file: " + outputFile.getAbsolutePath());
-        }
 
-        try {
+    /**
+     * Create a CRAMFileWriter on an output file.
+     *
+     * Note: does not honor factory setting for USE_ASYNC_IO.
+     * Read names and tags are preserved.
+     *
+     * @param header entire header. Sort order is determined by the sortOrder property of this arg.
+     * @param presorted  if true, SAMRecords must be added to the SAMFileWriter in order that agrees with header.sortOrder.
+     * @param outputFile where to write the output.  Must end with .sam, .bam or .cram.
+     * @param referenceFasta reference sequence file
+     * @return CRAMFileWriter
+     *
+     */
+    public CRAMFileWriter makeCRAMWriter(final SAMFileHeader header, final boolean presorted, final File outputFile, final File referenceFasta) {
+        return createCRAMWriter(header, presorted, outputFile, referenceFasta);
+    }
+
+    /**
+     * Create a CRAMFileWriter on an output file.
+     *
+     * Note: does not honor factory setting for USE_ASYNC_IO.
+     *
+     * @param header entire header. Sort order is determined by the sortOrder property of this arg.
+     * @param presorted  if true, SAMRecords must be added to the SAMFileWriter in order that agrees with header.sortOrder.
+     * @param outputFile where to write the output.  Must end with .sam, .bam or .cram.
+     * @param referenceFasta reference sequence file
+     * @return CRAMFileWriter
+     *
+     * Read names and tags are preserved.
+     */
+    private CRAMFileWriter createCRAMWriter(final SAMFileHeader header, final boolean presorted, final File outputFile, final File referenceFasta) {
```

Comment:
I think factoring out the extra 2 lines of configuration will at least make sure that no one will change one without the other. 

I do like the ability to use a stream instead of a file though.  The fact that we can't do that in many places is a real pain.  I guess you can always directly invoke the CRAMWriter  constructor if you want to write from a stream though and you're planning on ignoring the factory anyway.